poc/Cornelius.tbasic

```diff
@@ -117,180 +137,340 @@ function stirWorldMap()
 
 Function crossMap(toWhere as byte, borderPosition as DWord)
 
   local isCrossing as byte
-  Local maxLarge as Byte = 1 + myRiver.maxHalfWidth * 2
   Local fromTo, startPosition as tXY
-  Local minStretchLength, maxExpandPlace, actualExpand, fromToSelectSize, positionSelection, positionOffset as DWord
+  Local minStretchLength, maxExpandPlace, minExpandPlace, actualExpand, fromToSelectSize, maxSelectSize, positionSelection, positionOffset as DWord
 
   select case toWhere
     case %EAST
-
       for i as DWord = 1 to mapDimensions.x
 
         if map.isGameZone(i, borderPosition) Then
-        
           incr isCrossing
-          
         elseif isCrossing > 0 Then
-        
           if isCrossing > maxLarge Then
             isCrossing = 0
             Iterate for
           endif
 
-
           startPosition.x = i
           startPosition.y = borderPosition
-            
           getStretchLength(toWhere, startPosition, isCrossing, fromTo)
 
+          logger.write(%LT_ToConsole, "Crossed a " & str$(fromTo.y - fromTo.x) & " gameZones long stretch of a " & str$(isCrossing) & " gameZones wide river", Function_Name, %TAG_DEBUG)
+
           minStretchLength = 2 + isCrossing * 2 + myRiver.banksWidth * 4
-          
+
           if fromTo.y - fromTo.x >= minStretchLength Then
-          
-            fromToSelectSize = rnd(minStretchLength, Max(minStretchLength, Fix( 0.9 * (fromTo.y - fromTo.x - myRiver.banksWidth * 4))))
-            'printl "minStretchLength: " , minStretchLength , "select size: ", fromToSelectSize, "out of : ", str$(fromTo.y - fromTo.x)
+            maxSelectSize = Max(minStretchLength, Fix( myRiver.meander* (fromTo.y - fromTo.x - myRiver.banksWidth * 4)))
+            fromToSelectSize = rnd(minStretchLength, maxSelectSize)
+
+            logger.write(%LT_ToConsole, "Selected = " & str$(fromToSelectSize) & " out of [" & str$(minStretchLength) & ";" & str$(maxSelectSize) & "]" , Function_Name, %TAG_DEBUG)
+
             positionSelection = (fromTo.y - fromTo.x - fromToSelectSize)
-            
-            positionOffset = rnd(myRiver.banksWidth * 4, fix(PositionSelection/2))
-            
+            positionOffset = rnd(min(myRiver.banksWidth * 4,fix(PositionSelection/2)), fix(PositionSelection/2))
             fromTo.x += positionOffset
             fromTo.y = fromTo.x + fromToSelectSize
-
             maxExpandPlace = getExpandPlace(toWhere, fromTo, startPosition, isCrossing)
+            minExpandPlace = startPosition.x - isCrossing - 1 - 2 * myRiver.banksWidth
+
+            if maxExpandPlace < minExpandPlace Then
 
-            if maxExpandPlace < startPosition.x - isCrossing - 1 - 2 * myRiver.banksWidth Then
-            
-              actualExpand = fix(0.15 * (startPosition.x - isCrossing - 1 - maxExpandPlace))
+' actualExpand = fix(myRiver.meandering * (startPosition.x - isCrossing - 1 - maxExpandPlace))
+'  maxExpandPlace = rnd(maxExpandPlace + actualExpand, startPosition.x - isCrossing - 1 - actualExpand)
```

Comment:
for each directions, I commented out _actualExpand_ and _maxExpandPlace_ for debugging _refund_ and _widen_ linecodes